web/src/recent_view_ui.ts

```diff
@@ -852,40 +980,50 @@ function sort_comparator(a, b) {
     return -1;
 }
 
-function stream_sort(a, b) {
+function stream_sort(a: Row, b: Row): number {
     if (a.type === b.type) {
         const a_msg = message_store.get(a.last_msg_id);
+        assert(a_msg !== undefined);
         const b_msg = message_store.get(b.last_msg_id);
+        assert(b_msg !== undefined);
 
-        if (a.type === "stream") {
+        if (a_msg.type === "stream") {
+            assert(b_msg.type === "stream");
             const a_stream_name = stream_data.get_stream_name_from_id(a_msg.stream_id);
             const b_stream_name = stream_data.get_stream_name_from_id(b_msg.stream_id);
             return sort_comparator(a_stream_name, b_stream_name);
         }
+        assert(a_msg.type === "private");
+        assert(a_msg.display_reply_to !== undefined);
+        assert(b_msg.type === "private");
+        assert(b_msg.display_reply_to !== undefined);
```

Comment:
These extra asserts on `display_reply_to` seem like they reflect a problem with the `Message` type; it should be able to infer that the type implies that property.

Reply:
I didn't see any other instances of `display_reply_to !== undefined`, so hopefully it's actually just a quite simple change.